### PR TITLE
feat: pass MDX frontmatter title into page `<title>`

### PR DESCRIPTION
**Pull request by emyarod**
this PR passes the MDX frontmatter title into the `<Meta>` component HTML title element. not sure how configurable we want this to be, but this is an option for getting dynamic page titles in the format we're looking for in the IDL site

### CODE REVIEW

**Comment by vpicone** [comment]
@emyarod this looks great thanks. Sorry I was experimenting with the `www` site and might have caused the now deployment issue. Should be fixed with your next commit

Could you add optional description/keywords as well? I know brand center was asking for this.

**Comment by emyarod** [comment]
@vpicone just double checking, would optional description/keywords also be from MDX frontmatter?

**Comment by vpicone** [comment]
@emyarod yeah, that'd be ideal

**Comment by vpicone** [comment]
If they don't specify in frontmatter, it needs to be undefined so as to not override the one from siteMetadata

**Comment by emyarod** [comment]
local build was taking a while so I haven't had a chance to preview this but I wrote a few statements about the frontmatter. let me know if I should add more info @vpicone

File: packages/example/src/pages/guides/configuration.mdx [hunk 1]
```diff
@@ -148,14 +148,16 @@ To add a link to the bottom of each page that points to the current page source
 
 - `additionalFontWeights` – add support for additional Plex font weights. Don't forget to specify italics for the additional weights if needed.
 - `mdxExtensions` – change the file extensions processed by `gatsby-mdx` (default ['.mdx', '.md'])
+- `titleType` – the format of the title in your head. `site` for your site title, `page` for your individual page title, or `append` for "title – pageTitle" (default 'page')
```
Comment (alisonjoseph) [inline packages/example/src/pages/guides/configuration.mdx]: I think we need some examples here to make it easier to understand what each one means.

Comment (vpicone) [reply packages/example/src/pages/guides/configuration.mdx]: @alisonjoseph  added

**Comment by emyarod** [comment]
@vpicone yeah looks good to me. wasn't sure what kind of flexibility we were gonna offer WRT page title but this gives users more options. what's causing the lockfile changes though?

**Comment by vpicone** [comment]
@emyarod mdx had a patch bump while I was working on it, nothin major

**Comment by emyarod** [comment]
@vpicone that's fine was just wondering what prompted the change

**Pull request merged**